cpp/csp/python/PyAdapterManager.cpp

```diff
@@ -68,6 +68,19 @@ class PyAdapterManager : public AdapterManager
 static PyObject * PyAdapterManager_PyObject_starttime( PyAdapterManager_PyObject * self ) { return toPython( self -> manager -> starttime() ); }
 static PyObject * PyAdapterManager_PyObject_endtime( PyAdapterManager_PyObject * self )   { return toPython( self -> manager -> endtime() ); }
 
+static PyObject * PyAdapterManager_PyObject__engine_shutdown( PyAdapterManager_PyObject * self, PyObject * args, PyObject * kwargs)
+{
+    CSP_BEGIN_METHOD;
+
+    const char * msg;
+    if( !PyArg_ParseTuple( args, "s", &msg ) )
+        return NULL;
+
+    self -> manager -> rootEngine() -> shutdown( std::make_exception_ptr( TracebackStringException( msg ) ) );
```

Comment:
I wonder if we can make this work "correctly" by passing through the exception from python
Could we just create a PythonPassthrough exception that takes the type/value/traceback directly ( so a separate ctor in addition tot he one that fetches curr error ).  
when the engine on the main thread raises the PythonPassthrough, it should get restored and emit the original python exception, at least in theory

Reply:
Yes, I think that approach should work. Not sure if you saw my earlier comment here: https://github.com/Point72/csp/pull/312#discussion_r1664422606. If we can just create the right exception before we call `shutdown` then that's best.

Reply:
yeah I see that now, we should have followed through on that
i dont think you need to create any new object / store any dialect generic types.  Just create a PythonPassthrough exception

Reply:
Done